preshower planes calib conditions

amartelli opened the pull request:
to include ES single plane calibration in case of dead planes

Commit:
Validation toolket for HGCAL

Commit:
preshower planes calib conditions

perrotta committed:
Update HLT menus up to /dev/CMSSW_7_6_0/HLT/V27

cmsbuild commented:
A new Pull Request was created by  @amartelli  for CMSSW_8_0_X.

It involves the following packages:

RecoEcal/EgammaClusterAlgos
RecoParticleFlow/PFClusterTools
RecoParticleFlow/PFProducer


@cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please review it and eventually sign? Thanks.
@mmarionncern, @Sam-Harper, @argiro, @bachtis, @lgray this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

Following commands in first line of a comment are recognized

* **+1|approve[d]|sign[ed]**: L1/L2's to approve it
* **-1|reject[ed]**: L1/L2's to reject it
* **assign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to request signatures from other categories
* **unassign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to remove signatures from other categories
* **hold**: L1/all L2's/release manager to mark it as on hold
* **unhold**: L1/user who put this PR on hold
* **merge**: L1/release managers to merge this request
* **[@cmsbuild,] please test**: L1/L2 and selected users to start jenkins tests
* **[@cmsbuild,] please test with cms-sw/cmsdist#&lt;PR&gt;**: L1/L2 and selected users to start jenkins tests using externals from cmsdist

slava77 commented on RecoEcal/EgammaClusterAlgos/src/PFECALSuperClusterAlgo.cc:
```diff
@@ -373,23 +390,47 @@ buildSuperCluster(CalibClusterPtr& seed,
       for( auto i_ps = clustops.first; i_ps != clustops.second; ++i_ps) {
 	edm::Ptr<reco::PFCluster> psclus(i_ps->second);
+
+	const std::vector<reco::PFRecHitFraction> recH_Frac = psclus->recHitFractions();
```
``` const std::vector<reco::PFRecHitFraction>& recH_Frac ``` or even better yet ``` auto const& recH_Frac ```

slava77 commented on RecoEcal/EgammaClusterAlgos/src/PFECALSuperClusterAlgo.cc:
```diff
@@ -373,23 +390,47 @@ buildSuperCluster(CalibClusterPtr& seed,
 	switch( psclus->layer() ) {
 	case PFLayer::PS1:
 	  ps1_energies.push_back(psclus->energy());
+	  for(unsigned int iVec=0; iVec<recH_Frac.size(); ++iVec){
```
change to 
```C++
for (auto const& recH : recH_Frac){
ESDetId strip1 = recH.recHitRef()->detId();
```
same  for the loop below

slava77 commented on RecoParticleFlow/PFClusterTools/src/PFEnergyCalibration.cc:
```diff
@@ -801,8 +804,29 @@ PFEnergyCalibration::EcorrPS(double eEcal,double ePS1,double ePS2,double etaEcal
-  outputPS2=gammaprime*Alpha(etaEcal)*ePS2;
+
+  if(outputPS1 == 0 && outputPS2 == 0 && esEEInterCalib_ != 0){
+    //    std::cout << " >>> both working " << std::endl;
```
Please remove commented out code.
If you think they are needed, change to LogTrace or LogDebug

slava77 commented on RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc:
```diff
@@ -74,9 +79,11 @@ PFEGammaProducer::PFEGammaProducer(const edm::ParameterSet& iConfig,
-  std::shared_ptr<PFSCEnergyCalibration>  
-    thePFSCEnergyCalibration ( new PFSCEnergyCalibration(calibPFSCEle_Fbrem_barrel,calibPFSCEle_Fbrem_endcap,
-                                                         calibPFSCEle_barrel,calibPFSCEle_endcap )); 
+  // std::shared_ptr<PFSCEnergyCalibration>  
```
here and below, please remove the commented out code

slava77 commented on RecoParticleFlow/PFProducer/src/PFEGammaAlgo.cc:
```diff
@@ -2250,17 +2257,59 @@ buildRefinedSuperCluster(const PFEGammaAlgo::ProtoEGObject& RO) {
+      for( auto i_ps = psclusters.begin(); i_ps != psclusters.end(); ++i_ps) {
+        const PFClusterRef&  psclus = i_ps->first->clusterRef();
+
+        const std::vector<reco::PFRecHitFraction> recH_Frac = psclus->recHitFractions();
```
see comments in  RecoEcal/EgammaClusterAlgos/src/PFECALSuperClusterAlgo.cc ,
the same applies here

slava77 commented:
@amartelli if some output changes are expected,
please provide a brief description of expected changes at physics level and add a link to slides or plots (assuming this was presented in the DPG meeting)

Commit:
preshower planes calib conditions

Commit:
preshower planes calib conditions

cmsbuild commented:
Pull request #12929 was updated. @cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please check and sign again.

amartelli commented:
Changes are implemented. 
No output changes in the energy reconstruction are expected, if the proper set of tags is used (all this was discussed and approved at the DPG @emanueledimarco)

slava77 commented:
If there was a presentation in the DPG, please add a link to the presentation in the PR description (at the top of the PR page)

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

emanueledimarco commented:
@slava77 here is the presentation (done today): https://indico.cern.ch/event/482171/contribution/4/attachments/1210846/1766046/arm_EScalib_updateCode.pdf

cmsbuild commented:
-1 
Tested at: 9fb5e277748d1ceb36ebb2a29a40de50a0b0fa4b
 I found an error when building: 
 
 <pre>Assembler messages:
Error: can't read from {standard input}: Bad address
/bin/sh: line 1: 21848 Segmentation fault      /bin/echo ">> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-12-2300/src/RecoEcal/EgammaClusterProducers/src/InterestingTrackEcalDetIdProducer.cc "
gmake: *** [tmp/slc6_amd64_gcc493/src/RecoEcal/EgammaClusterProducers/src/RecoEcalEgammaClusterProducers/InterestingTrackEcalDetIdProducer.o] Error 139
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-12-2300/src/RecoEcal/EgammaClusterProducers/src/IslandClusterProducer.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-12-2300/src/RecoEcal/EgammaClusterProducers/src/CleanAndMergeProducer.cc:213:1: fatal error: error writing to -: Broken pipe
 }
 ^
compilation terminated.
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-12-2300/src/RecoEcal/EgammaClusterProducers/src/Multi5x5ClusterProducer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-12-2300/src/RecoEcal/EgammaClusterProducers/src/Multi5x5SuperClusterProducer.cc 
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12929/10484/summary.html

slava77 commented:
@cmsbuild please test
the last tests have failed apparently due to an infrastructure error (actually, a segfault during compilation)

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9fb5e277748d1ceb36ebb2a29a40de50a0b0fa4b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12929/10487/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12929/10487/summary.html

slava77 commented:
On 1/13/16 9:31 AM, amartelli wrote:
> Changes are implemented.
> No output changes in the energy reconstruction are expected, if the
> proper set of tags is used (all this was discussed and approved at the
> DPG @emanueledimarco <https://github.com/emanueledimarco>)
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12929#issuecomment-171372920>.
> 

Hi Arabella and Emanuele

Based on jenkins tests there are quite a few changes

e.g.
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_8_0_X_2016-01-13-1100+12929/11976/validateJR/all_OldVSNew_RunSinglePh2015Dwf134p911/all_OldVSNew_RunSinglePh2015Dwf134p911c_log10recoSuperClusters_particleFlowSuperClusterECAL_particleFlowSuperClusterECALEndcapWithPreshower_reRECO_obj_energy.png
shows about 10% increase in the  number of clusters
This is in SinglePhoton Run2015D processing

A similar increase shows up in ttbar MC (PU35) as well
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_8_0_X_2016-01-13-1100+12929/11976/validateJR/all_OldVSNew_TTbarPUwf25202p0/all_OldVSNew_TTbarPUwf25202p0c_log10recoSuperClusters_particleFlowSuperClusterECAL_particleFlowSuperClusterECALEndcapWithPreshower_RECO_obj_energy.png

It's unclear if this is from something related to dead planes (do we
have any in MC?)
or from changes in the PFEnergyCalibration formulas
(including intercalibration constants).
https://github.com/cms-sw/cmssw/pull/12929/files#diff-cb4c25790ee3eaa0caf6024baa44b158R811


Does this PR need a new GT?
(your message suggests that there should be no changes).
If a GT update is needed, it better be combined with this PR.

If the GT is OK and some changes are expected, do you have any higher
level plots,
e.g. EE-EE Zee resolution or just particle gun showing at pfCandidate or
gedElectron/Photon level that we get an improvement after your changes.
(the slides linked by Emanuele were pointing only to resolutions
compared with dead and live planes,
not before and after this PR).

amartelli commented:
Hi Slava, 
the correct tag to be used with data is ESIntercalibConstants_LG_data_offline_default available in PREP, but I don't know if the standard integration tests can be done with an es_prefer of a tag in PREP or you need something else.

amartelli commented:
Two tags are needed ESIntercalibConstants_LG_data_offline_default at the moment in PREP and ESIntercalibConstants_V03_offline in PROD.
Sorry the previous incomplete comment

slava77 commented:
I guess in this case an es_prefer will work best for testing.
This PR will need to be combined with a new GT including these new payloads to be integrated in the release.
@mmusich @diguida

amartelli commented:
Both tags are now available  in PROD with the names ESIntercalibConstants_LG_offline_data_default and ESIntercalibConstants_V03_offline

diguida commented:
@franzoni to watch

mmusich commented:
@slava77 let's wait https://github.com/cms-sw/cmssw/pull/12935 is merged. Or is it urgent?

mmusich commented:
please @amartelli put the tags you prepared in queue

mmusich commented:
By the way, why 2 versions of the ESIntercalibConstants?

amartelli commented:
It's a change in the name from PREP to PROD, a mistake. The content is the same
What do you mean with "put the tags in queue"? never done before...

thanks, Arabella

mmusich commented:
@amartelli https://cms-conddb.cern.ch/cmsDbBrowser/request/Prod, then you need to choose the scenario and workflow, that I understand will be run1 and run2 data.  
I am confused by this statement:

```
Two tags are needed ESIntercalibConstants_LG_data_offline_default at the moment in PREP and ESIntercalibConstants_V03_offline in PROD.
``
are both of them needed?

slava77 commented:
On 1/14/16 2:41 AM, Marco Musich wrote:
> @slava77 <https://github.com/slava77> let's wait #12935
> <https://github.com/cms-sw/cmssw/pull/12935> is merged. Or is it urgent?
> 

I don't think it's extremely urgent, but then it depends how long we'll
have to wait for 12935.
If someone can prepare the es_prefer-related commands to call, I can run
tests in parallel.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12929#issuecomment-171607811>.
>

mmusich commented:
@slava77 I would prefer if the tags are in queue. At that point a GT candidate is a click away, and we'll avoid es_prefer around

slava77 commented:
On 1/14/16 2:51 AM, Marco Musich wrote:
> @amartelli <https://github.com/amartelli>
> https://cms-conddb.cern.ch/cmsDbBrowser/request/Prod, then you need to
> choose the scenario and workflow, that I understand will be run1 and
> run2 data.
> 

Based on tests made with the old payloads, there are changes in MC as well.
So, it sounds like MC needs to change as well.

@amartelli  please confirm

> I am confused by this statement:
> 
> |Two tags are needed ESIntercalibConstants_LG_data_offline_default at the
> moment in PREP and ESIntercalibConstants_V03_offline in PROD. `` are
> both of them needed? |
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12929#issuecomment-171610730>.
>

amartelli commented:
You are right, ESIntercalibConstants_V03_offline is ok and corresponds to the record ESIntercalibConstantsRcd. 
ESIntercalibConstants_LG_data_offline_default has the good content and a bad name, it corresponds to the record ESEEIntercalibConstants.
both these are needed, can ask to change the name and put the good ones in queue.

thanks

amartelli commented:
yes also a set of tags for MC is ready, I can proceed putting them in queue as well

slava77 commented:
On 1/14/16 2:56 AM, Marco Musich wrote:
> @slava77 <https://github.com/slava77> I would prefer if the tags are in
> queue. At that point a GT candidate is a click away, and we'll avoid
> es_prefer around

Hi Marco,

If the queue has changes in addition to what's in the fixed snapshot GT,
then making incremental comparisons is pretty difficult without es_prefer.
I guess one way is to figure out the insertion times and then modify the
snapshot times
in the tests.
What do you think?

> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12929#issuecomment-171612522>.
>

mmusich commented:
@slava77 my proposal is to create a fixed snapshot candidate out of the queue (once the desired tags are available) that can be tested against this PR. As soon as we are all happy I'll approve the tag changes in the queue and create the GTs to be merged together with this PR.  The next set of incremental changes will include the new ES conditions (already approved in queue). Do we loose in simplicity with respect to the es_prefer model?

slava77 commented:
On 1/14/16 3:58 AM, Marco Musich wrote:
> @slava77 <https://github.com/slava77> my proposal is to create a fixed
> snapshot candidate out of the queue (once the desired tags are
> available) that can be tested against this PR. As soon as we are all
> happy I'll approve the tag changes in the queue and create the GTs to be
> merged together with this PR. The next set of incremental changes will
> include the new ES conditions (already approved in queue). Do we loose
> in simplicity with respect to the es_prefer model?

juggling snapshot times is easier than es_prefers.
So, let's move along with your proposal.



> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12929#issuecomment-171625815>.
>

diguida commented:
@amartelli @mmusich sorry to chime in.
For 2015 re-reco we included in the GT [`76X_dataRun2_v16`](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/76X_dataRun2_v16) a new tag `ESIntercalibConstants_Run1_Run2_V04_offline` where the data/mc energy scale disagreement was corrected with a scale factor: see [summary talk at AlCaDB, slide 4](https://indico.cern.ch/event/465221/contribution/6/5/attachments/1201423/1748538/MoCa_PhysicsWeek081215.pdf) and [presentation at MoCa meeting](https://indico.cern.ch/event/464677/contribution/1/attachments/1195171/1736894/ES_2015intercalibration.pdf). ES experts announced that they were working on correcting the PF clustering for ES in order to fix this. #12636 and #12664 for GT integration.

IIUC now, this PR, as expected, changes the PF ES clustering including ES single plane calibration in case of dead planes: from the conditions point of view, this means loading two new records `ESEEIntercalibConstantsRcd` (see https://github.com/cms-sw/cmssw/pull/12929/files#diff-35b1d0d0535599d6d5df02ae48e146e9R207) and `ESChannelStatusRcd` (see https://github.com/cms-sw/cmssw/pull/12929/files#diff-35b1d0d0535599d6d5df02ae48e146e9R211) and consuming the latter in the SC algo (see e.g. https://github.com/cms-sw/cmssw/pull/12929/files#diff-35b1d0d0535599d6d5df02ae48e146e9R401), the former in the PF energy calibration (see https://github.com/cms-sw/cmssw/pull/12929/files#diff-cb4c25790ee3eaa0caf6024baa44b158R807).

As a consequence, this triggers a change in MC conditions, which touches the data/mc energy scale, thus requiring also new data calibration.

Is this correct?

Given that this strategy was proposed by AlCa, it would've been kind to keep us... ehm... them in the loop.

@slava77 I think the candidate is the right solution:
* the record/tag changes are easily understandable via CLI or web diff
* the changes can be easily traced also in terms of IOVs
* if the new conditions are ok, the tag is approved in the queue and a new GT is created and set as reference for IB testing (i.e. `autoCond`).

amartelli commented:
@diguida, your summary is correct.
The tags for the record ESEEIntercalibConstantsRcd are now properly named (ESEEIntercalibConstants_LG_offline_data_default for data and ESEEIntercalibConstants_LG_mc_default for mc).
The conditions for data require to use the ESIntercalibConstants_V03_offline which does not contain the  data/mc energy scale correction (now included in ESEEIntercalibConstants_LG_offline_data_default). 
Ask confirmation @cmkuo about the use of the default  ESIntercalibConstantsRcd tag for MC and @emanueledimarco @shervin86 about ESChannelStatusRcd and other eventual conditions.

thanks a lot

diguida commented:
@mmusich for what matters, we are in perfect agreement :-).
Concerning testing of new conditions:
* for module testing by developers, `es_prefer` just works (reminder: if you customise the GT using `toGet`, the snapshot time of the change is automatically set to infinity https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/CondCore/ESSources/plugins/CondDBESSource.cc#L152-L173)
* for integration testing: the GT should be the entry point for all conditions (separation of concerns and more maintainable code).
My 2 cents

mmusich commented:
@amartelli as @diguida points out actually the correct tag to be used is `ESIntercalibConstants_Run1_Run2_V03_offline` since `ESIntercalibConstants_V03_offline` misses the run-I history https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/tags/ESIntercalibConstants_Run1_Run2_V03_offline/ESIntercalibConstants_V03_offline

mmusich commented:
for the record: see discussion here https://github.com/cms-sw/cmssw/pull/12636#issuecomment-161598141

diguida commented:
@mmusich thanks for re-spotting the issue of the original tags: I also linked the integration history of the two GT PR in https://github.com/cms-sw/cmssw/pull/12929#issuecomment-171629160.

@amartelli thanks for the further clarification. So, with this PR you also improved not only the consuming of conditions for PF ES clustering (you are now loading the `ESEEIntercalibConstantsRcd`), but also the performance of the PF itself, as you are not relying anymore on some hardcoded numbers, but on a condition payload shaping the data/mc energy scale correction.

In summary: the condition modifications to be tested here are:
* for MC: `ESEEIntercalibConstantsRcd -> ESEEIntercalibConstants_LG_mc_default`
* for data: `ESIntercalibConstantsRcd -> ESIntercalibConstants_Run1_Run2_V03_offline` (already requested in queue in previous iteration) and `ESEEIntercalibConstantsRcd -> ESEEIntercalibConstants_LG_offline_data_default`.

As @mmusich pointed out, please submit the `ESEEIntercalibConstants` in the corresponding queues.

mmusich commented:
@amartelli @emanueledimarco since I haven't seen any further activity I took the liberty to queu:
   * `ESEEIntercalibConstants_LG_mc_default` for all the Run2 MC scenarios (asymptotic, HI, startup, design)
   * `ESEEIntercalibConstants_LG_offline_data_default` for data run2

Questions:
   * shall we update also the Run1 GTs with `ESEEIntercalibConstants_LG_mc_default`?
   * what about the tag consumed at HLT `ESEEIntercalibConstants_V02_hlt` and `	ESIntercalibConstants_V05_hlt[cms_orcon_prod/CMS_COND_39X_PRESHOWER]`?

amartelli commented:
thanks for this.

Concerning your questions, I need information on how these are used at the HLT @cmkuo

amartelli commented:
@mmusich, relying on info from @cmkuo: 
- run1 GTs can be updated for both data (ESEEIntercalibConstants_LG_mc_default) and  MC (ESEEIntercalibConstants_LG_mc_default)
- also the tags consumed at the HLT need to be consistent with the offline ones, for both records ESEEIntercalibConstantsRcd and ESIntercalibConstantsRcd

thanks

cschomak committed:
Geometry comparison: Plotting of y-range improved and printPng and printOnlÃyGlobal options added

slava77 commented:
I suppose, GT candidate is still not out yet.
Is this correct?

mmusich commented:
@slava77 
for testing purposes you can use:
'''
'run2_data' : '80X_dataRun2_Candidate_2016_01_19_19_25_45',
'run2_mc' : '80X_mcRun2_asymptotic_Candidate_2016_01_19_19_28_15'
'''
let me know if you need more of them (for other workflows).

slava77 commented:
@mmusich 
for completeness run1_data and run1_mc would be useful.
I can start with the run2_\* you pointed me to

mmusich commented:
@slava77 I added a suitable `run1_mc` key in the matrix above. `run1_data` is == to `run2_data` since a couple of updates since we are dealing with differences using the IOV mechanism.

mmusich commented:
@tocheng something you might want to watch

Commit:
improve naming of hlt tracking sequences

Commit:
Soureek updating TOP trigger Path names accrosing to CMSHLT-666 JIRA ticket

Martin-Grunewald committed:
Rebase to 800pre5 and new test input files

Martin-Grunewald committed:
Fix HIon

Martin-Grunewald committed:
Accumulated HLT update

Commit:
clean up FastSimulation/HighLevelTrigger

Martin-Grunewald committed:
Move to 800 area in ConfDB

Commit:
Merged refs/pull/13032/head from repository cms-sw

Commit:
move hltPixelTracks to HLTBeginSequence

slava77 commented:
@mmusich 
I finally got around to testing the candidate GT from https://github.com/cms-sw/cmssw/pull/12929#issuecomment-172959784
It looks like it is still appropriate to test this in 800pre5.

I notice, however:
https://cms-conddb-dev.cern.ch/browser/#diff/Prod/gts/80X_mcRun2_asymptotic_v1/80X_mcRun2_asymptotic_Candidate_2016_01_19_19_28_15
the MC GT has a change only in ESEEIntercalibConstantsRcd

https://cms-conddb-dev.cern.ch/browser/#diff/Prod/gts/80X_dataRun2_v2/80X_dataRun2_Candidate_2016_01_19_19_25_45
the data GT has a change in ESEEIntercalibConstantsRcd and in ESIntercalibConstantsRcd.
For the latter, the version number is smaller: ESIntercalibConstants_Run1_Run2_V04_offline in v2 and ESIntercalibConstants_Run1_Run2_V03_offline in the candidate: payloads in IOVs starting from 246908 are different (v03 has insertion time 2015-12-03 11:58:09 and V04 has insertion time 2015-12-03 11:55:54). Is this intended? Is the update of ESIntercalibConstantsRcd in data a part of this PR needs?

Commit:
Updated singletopHLTEventDQM_cfi.py file under DQMOffline/Trigger/python/ to correct the errors obtained by tests triggered by jenkins

amartelli commented:
Yes the update of ESIntercalibConstantsRcd in data is needed. V03 was already prepared in DB in view of this PR and V04 was a temporary solution while awaiting this PR.
I confirm that the MC changes are only for ESEEIntercalibConstantsRcd

mmusich commented:
@amartelli thanks for the confirmation
@slava77 you might want to have a look also to the summary here: https://github.com/cms-sw/cmssw/pull/12929#issuecomment-173937189

lgray committed:
fix baseline problems in the hexagonal geometry, still encounter invalid detids

lgray committed:
rebasing cruft

slava77 commented:
Looking at the results, I'm not sure all are expected.
I used 800pre5 with this PR+edits to autoCond to pick up the new GTs:
https://github.com/cms-sw/cmssw/compare/CMSSW_8_0_0_pre5...slava77:CMSSW_8_0_0_pre5/sign663/pull12929-9fb5e27-GT011919h
more specifically https://github.com/cms-sw/cmssw/commit/94e0a92312461a52520fe36deee2ff3b9715d4e7

reco::Photons "photons" collection appear broken, while gedPhotons are essentially unchanged

1332 H->aa workflow
![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_log10recosuperclusters_correctedmulti5x5superclusterswithpreshower__reco_obj_energy](https://cloud.githubusercontent.com/assets/4676718/12530327/e4c5455e-c190-11e5-938c-f635028feed6.png)
![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_recophotons_photons__reco_obj_e5x5](https://cloud.githubusercontent.com/assets/4676718/12530335/1e4979e4-c191-11e5-8248-f01d0970adf1.png)
![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_recophotons_photons__reco_obj_energy](https://cloud.githubusercontent.com/assets/4676718/12530336/28960c1e-c191-11e5-8089-c9961404e7a9.png)
![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_recophotons_photons__reco_obj_energycorrections_phoecalenergy](https://cloud.githubusercontent.com/assets/4676718/12530342/6da76ff0-c191-11e5-9af1-83506285187c.png)
![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_recophotons_photons__reco_obj_energycorrections_scecalenergy](https://cloud.githubusercontent.com/assets/4676718/12530344/86c44f44-c191-11e5-8fb1-d2bf92c90e55.png)
the photon's energy seems substantially different from what is happening in the cluster shape energies (e5x5)

![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_recophotons_photons__reco_obj_et](https://cloud.githubusercontent.com/assets/4676718/12530346/93a69348-c191-11e5-8b91-00c73d6c2c6e.png)
![all_sign663-pass-94e0a92vsorig_h125gg13tevwf1332p0c_recophotons_photons__reco_obj_eta](https://cloud.githubusercontent.com/assets/4676718/12530347/97dfd168-c191-11e5-8952-2943bacf4628.png)

Ignoring collections with the following labels "_photons_\|_reduced\|_correctedMulti5x5SuperClustersWithPresh", the remaining changes are
- in HI data and in 2017 MC ... expected since I didn't update the GTs here (ignore for now, but note that you need to update GTs here)
-  in run1 data: here changes are pretty small and I'm guessing they show up from actual presence of 
dead ES planes
- in run2 data: here changes are larger and start from the ES rechits. This, I suppose is from differences in ESIntercalibConstantsRcd between the baseline and the candidate, which I would rather did not have in the comparisons related purely to the reco code parts of this PR.
e.g. in 134.911
![all_sign663-pass-94e0a92vsorig_runsingleph2015dwf134p911c_log10ecalrechitssorted_ecalpreshowerrechit_ecalrechitses_rereco_obj_obj_energy](https://cloud.githubusercontent.com/assets/4676718/12530425/197d190e-c194-11e5-93bc-ca33902de1ad.png)
![all_sign663-pass-94e0a92vsorig_runsingleph2015dwf134p911c_log10recopfclusters_particleflowclusterps__rereco_obj_energy](https://cloud.githubusercontent.com/assets/4676718/12530439/5bd523fa-c194-11e5-8de5-373dc9353225.png)
in the two above it is more likely that the cluster energy changes due to the hit energy change rather than from the cluster calibration issues

Commit:
ES planes conditions for old reco

Commit:
ES planes conditions for old reco

cmsbuild commented:
Pull request #12929 was updated. @cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please check and sign again.

amartelli commented:
also the standard reconstruction has been updated, given the changes spotted in reco::Photons.

lgray committed:
fix unit test for square geom

slava77 commented:
@amartelli  please clarify if the behavior in run2 data processing --changes in the ES hit energies -- is expected, while in MC the hit energies are not changing.

amartelli commented:
@slava77 record ESIntercalibConstantsRcd has been changed for data only and this record contains the calibration of the rechit energy: so I confirm that what you reported is expected.
thank you

slava77 commented:
@amartelli  is the change in ESIntercalibConstantsRcd required for this PR code update or is it just a part of other developments in AlCa?
As observed in MC, I understood that the code update in this PR required a matching update to the calibration payloads in order to keep the physics output essentially unchanged.
It seems to me that an update in ESIntercalibConstantsRcd is rather tangential, but may be I missed something.
Please clarify.

Thank you.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: e89d23510ae6adcfa967ee6d1875610bd5c3ce9a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12929/10691/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12929/10691/summary.html

Commit:
CalibCalorimetry/CaloTPG: definite static constexpr float members

The patch resolves issue with Clang compiler.
N3690 (should be C++11 standard) and latest draft N4567, 9.4.2/3

...
The member shall still be defined in a namespace scope if it
is odr-used (3.2) in the program and the namespace scope definition
shall not contain an initializer.
9.4.2/4 talks about how const static data members are being handled.

Also standard says that no diagnostic is required by compiler.

    src/CalibCalorimetryCaloTPG/CaloTPGTranscoderULUT.o: In function aloTPGTranscoderULUT::setup(HcalLutMetadata const&, HcalTrigTowerGeometry const&)':
    src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:(.text+0x168b): undefined reference to aloTPGTranscoderULUT::LSB_HBHE'
    src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:(.text+0x16b6): undefined reference to aloTPGTranscoderULUT::LSB_HF'

Before:

    $ nm -a ../lib/slc6_amd64_gcc493/libCalibCalorimetryCaloTPG.so | wc -l
    153

After:

    $ nm -a ../lib/slc6_amd64_gcc493/libCalibCalorimetryCaloTPG.so | wc -l
    155

    $ nm -a ../lib/slc6_amd64_gcc493/libCalibCalorimetryCaloTPG.so | c++filt | grep -E '(LSB_HF|LSB_HBHE)'
    000000000000871c R CaloTPGTranscoderULUT::LSB_HF
    0000000000008718 R CaloTPGTranscoderULUT::LSB_HBHE

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

lgray committed:
update to consumes

amartelli commented:
@slava77 the change of the ESIntercalibConstantsRcd is required for this PR, for what concerns the data-tag, while the tag for MC is unchanged: this to remove from the ESIntercalibConstants_tag_for_data the data/mc scaling factors that were introduced at the end of 2015 as a temporary solution and that aer now available in a different record.

Commit:
hard-code L1 et-sum eta boundaries

(cherry picked from commit 59ae76ae15db7c06a6266e9bd81687f53cd9756b)

Martin-Grunewald committed:
Merged HltUpdatesFor80X from repository cms-tsg-storm

Martin-Grunewald committed:
Cleanup

Martin-Grunewald committed:
HLT updates

cmsbuild committed:
Merge pull request #13032 from lveldere/fastsim-hltcfg

Fastsim: cleanup HLT configuration

cmsbuild committed:
Merge pull request #13038 from lgray/fix_hex_geom

Fixing HGCal Hexagonal Geometry Round 1

cmsbuild committed:
Merge pull request #13044 from davidlt/fix-clang-CaloTPGTranscoderULUT

CalibCalorimetry/CaloTPG: definite static const float members

cmsbuild committed:
Merge pull request #13046 from lgray/Run2-hcx62

Run2-hcx62 Validation toolkit for HGCAL

cmsbuild committed:
Merge pull request #12986 from cschomak/CMSSW_8_0_0_pre2_GeometryComparison

Geometry comparison: Plotting of y-range improved and print options added

cmsbuild committed:
Merge pull request #13027 from Soureek89/Soureek_Jan21_TriggerPathNameChange

Soureek updating TOP trigger Pathnames according to CMSHLT-666

cmsbuild committed:
Merge pull request #13048 from richard-cms/L1Stage1-HI-Fix-ETT-80X

Fix Stage-1 L1 HI ETT algorithm

cmsbuild committed:
Merge pull request #13037 from cms-tsg-storm/HltUpdatesFor80X

HLT development updates for 80X

cmsbuild committed:
Merge pull request #13052 from smuzaffar/ib-clang-static-analysis-fix

use the correct classes.h path (no duplicate src/src)

slava77 commented:
Looking at https://github.com/cms-sw/cmssw/pull/12929	e89d235 and running with the candidate GTs, I see there are only small differences in data (more significant in run2 due to the change in ESIntercalibConstantsRcd). The issues in the old photon and multi5x5 clusters are gone.
So, I think that the new payloads are good and they can go to the actual GT. (updates for phase1_2017_design are needed as well).

Once there is a GT, either this PR will need to include changes to autoCond.py or another PR will need to merge this PR.

mmusich commented:
@slava77 I'll make sure this will happen in the next hours. Thanks

mmusich committed:
Introducing new ES calibrations tags due to updated PF ES clustering

mmusich commented:
@amartelli the new Global Tags are availble: https://github.com/cms-sw/cmssw/compare/CMSSW_8_0_X...mmusich:GTs_80X_v2?expand=1
Would you mind pulling from my branch to get them?

Commit:
Merge pull request #2 from mmusich/GTs_80X_v2

G ts 80 x v2

cmsbuild commented:
Pull request #12929 was updated. @perrotta, @smuzaffar, @civanch, @diguida, @lveldere, @cvuosalo, @fwyzard, @ianna, @mdhildreth, @Martin-Grunewald, @ssekmen, @cmsbuild, @Dr15Jones, @cerminar, @deguio, @slava77, @franzoni, @mmusich, @vanbesien, @mulhearn, @davidlange6 can you please check and sign again.

slava77 commented:
the update was supposed to have been a cherry-pick, not a merge.
The PR needs a rebase or the last merge redone.

mmusich commented:
@slava77 I don't know if this is by any means more convenient but I took the liberty to push to my branch the relevant commits from @amartelli. The PR nr is #13074

slava77 commented:
@mmusich 
we can continue in 13074, although staying here in the same PR may be more convenient especially if changes in the GT are otherwise minimal.

slava77 commented:
@amartelli 
please close this PR, unless you manage to redo the last merge
